From bdf6e5fcfd0c8bc956545ebf2855de36cab855b5 Mon Sep 17 00:00:00 2001
From: David Herrmann <dh.herrmann@gmail.com>
Date: Fri, 25 Aug 2017 15:37:00 +0200
Subject: [PATCH 1/3] test: don't hardcode 64bit long

Lets not rely on 64bit longs in our tests. We run just fine on 32bit
platforms, so lets make our tests run there as well.

Reported-by: Jan Engelhardt
Signed-off-by: David Herrmann <dh.herrmann@gmail.com>
---
 src/test-macro.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/test-macro.c b/src/test-macro.c
index ad73c03..1e2fdf7 100644
--- a/src/test-macro.c
+++ b/src/test-macro.c
@@ -279,7 +279,7 @@ static void test_misc(int non_constant_expr) {
                 static_assert(C_DECIMAL_MAX(char) == 4, "");
                 static_assert(C_DECIMAL_MAX(signed char) == 4, "");
                 static_assert(C_DECIMAL_MAX(unsigned char) == 4, "");
-                static_assert(C_DECIMAL_MAX(unsigned long) == 21, "");
+                static_assert(C_DECIMAL_MAX(unsigned long) == (sizeof(long) == 8 ? 21 : 11), "");
                 static_assert(C_DECIMAL_MAX(unsigned long long) == 21, "");
                 static_assert(C_DECIMAL_MAX(int32_t) == 11, "");
                 static_assert(C_DECIMAL_MAX(uint32_t) == 11, "");
-- 
2.14.2

